lint/src/main/java/com/infinum/jsonapix/lint/JsonApiXCodeDetector.kt

```diff
@@ -0,0 +1,65 @@
+package com.infinum.jsonapix.lint
+
+import com.android.tools.lint.client.api.UElementHandler
+import com.android.tools.lint.detector.api.Category
+import com.android.tools.lint.detector.api.Detector
+import com.android.tools.lint.detector.api.Implementation
+import com.android.tools.lint.detector.api.Issue
+import com.android.tools.lint.detector.api.JavaContext
+import com.android.tools.lint.detector.api.Scope
+import com.android.tools.lint.detector.api.Severity
+import org.jetbrains.uast.UAnnotated
+import org.jetbrains.uast.UAnnotation
+import org.jetbrains.uast.UElement
+import org.jetbrains.uast.getParentOfType
+import java.util.EnumSet
+
+private const val JSON_API_X = "com.infinum.jsonapix.annotations.JsonApiX"
+private const val JSON_API_X_META = "com.infinum.jsonapix.annotations.JsonApiXMeta"
+private const val SERIALIZABLE = "kotlinx.serialization.Serializable"
+
+@Suppress("UnstableApiUsage")
+class JsonApiXCodeDetector : Detector(), Detector.UastScanner {
+
+    override fun getApplicableUastTypes(): List<Class<out UElement>> = listOf(UAnnotation::class.java)
+
+    override fun createUastHandler(context: JavaContext): UElementHandler {
+        return object : UElementHandler() {
+
+            override fun visitAnnotation(node: UAnnotation) {
+                if (node.qualifiedName == JSON_API_X || node.qualifiedName == JSON_API_X_META) {
```

Comment:
`JsonApiX`, `JsonApiXMeta` and `JsonApiXLinks` all have to also be annotated with the `Serializable` as well. Can we add `JsonApiXLinks` to the check and also update our issue to use the correct name in the error message?
For example`"@JsonApiX must be combined with @Serializable"` does not apply to errors with `JsonApiXMeta` and `JsonApiXLinks`. Maybe issue can take a parameter to determine for which annotation the warning is. If not, we can create 3 separate issues

Reply:
My bad, forgot about links. 

In this case we can just do a simple when and act accordingly. Update coming up 👍